src/action/cart.js

```diff
@@ -0,0 +1,52 @@
+import {
+  ADD_TO_CART,
+  CHANGE_QTY_PRODUCT,
+  CHANGE_QTY_PRODUCT_DOWN,
+  CHANGE_QTY_PRODUCT_UP,
+  SET_QTY_ENTER_VALUE
+} from "../constants";
+
+export const addToCart = idProduct => {
+  return (dispatch, getState) => {
+    const { cart } = getState();
+    const { qtyProduct } = cart;
+    dispatch({ type: ADD_TO_CART, payload: { idProduct, qtyProduct } });
+  };
+};
+
+export const changeQtyProduct = controlQty => {
+  return (dispatch, getState) => {
+    const { cart } = getState();
+    const { qtyProduct } = cart;
+    if (controlQty === CHANGE_QTY_PRODUCT_UP) {
+      const newQty = parseInt(qtyProduct + 1);
+      if (qtyProduct + 1 <= 300) {
+        dispatch({ type: CHANGE_QTY_PRODUCT, payload: newQty });
+      }
+    }
+
+    if (controlQty === CHANGE_QTY_PRODUCT_DOWN) {
+      const newQty = parseInt(qtyProduct - 1);
+      if (qtyProduct - 1 > 0) {
+        dispatch({ type: CHANGE_QTY_PRODUCT, payload: newQty });
+      }
+    }
+  };
+};
+
+export const changeEnteredValue = enteredValue => {
```

Comment:
action have bad name `changeEnteredValue` what value? where you change it? where you entered it?